core/key.go

```diff
@@ -41,18 +42,18 @@ const (
 )
 
 const (
-	COLLECTION                        = "/collection/names"
-	COLLECTION_SCHEMA                 = "/collection/schema"
-	COLLECTION_SCHEMA_VERSION         = "/collection/version/v"
-	COLLECTION_SCHEMA_VERSION_HISTORY = "/collection/version/h"
-	COLLECTION_INDEX                  = "/collection/index"
-	SCHEMA_MIGRATION                  = "/schema/migration"
-	SCHEMA_VERSION                    = "/schema/version"
-	SEQ                               = "/seq"
-	PRIMARY_KEY                       = "/pk"
-	DATASTORE_DOC_VERSION_FIELD_ID    = "v"
-	REPLICATOR                        = "/replicator/id"
-	P2P_COLLECTION                    = "/p2p/collection"
+	COLLECTION                     = "/collection/value"
```

Comment:
suggestion: I think `COLLECTION_NUMBER = "/collection/number"` would be a bit more representative in this case. I had mentionned previously that we could have `collection.Index` instead of the current `collection.ID` so that we could use `collection.ID` for the  global ID. The problem is that it would then create confusion with the secondary indexes. This is why I'm suggesting to use `collection.Number`.

suggestion: I think having `COLLECTION_ID = "/collection/id"` would be important to reference the collection with it's global ID.

Reply:
> The problem is that it would then create confusion with the secondary indexes. This is why I'm suggesting to use collection.Number.

Did I misread, or did you edit? :sweat_smile: 

I missed this when replying, `number` feels a bit strange, but I have no better suggestions yet.

Reply:
> thought: colllection.Index will also be a confusing name for that reason. Maybe we can find something else when we get round to looking at that.

`collection.Number`. would be fine as noted above.

Reply:
> Then again, I think this key prefix should align with that property, which means it should either be left as collection/id for now, or we need to solve the collection.ID property name problem here/now. Worth bearing in mind is that changing this is a breaking change, so it is better if we don't change it every other day.

I really think `collection.ID` should be used for the global ID.

Reply:
> Did I misread, or did you edit? 😅

I did not edit :)

Reply:
Discussed in stand-up, we are undecided but we set a deadline for noon Tuesday Ottawa time to pick something.  `Number` is still a bit weird, but otherwise received little criticism (unlike `LocalID`, which could be confusing for a few reasons.  If no consensus is reached by the deadline I will probably rename to `Number`.

- [ ] Wait/discuss/rename to `Number

Reply:
Leaving as is for now, hopefully this can be removed in https://github.com/sourcenetwork/defradb/issues/1085

Reply:
Can we do something about the actual key string and though? Value isn't representative of the key.

Reply:
Ah, true sorry - I'll change it to `collection/id`

- [ ] `collection/id`